### PR TITLE
Adds a message when trying to move while buckled

### PR DESCRIPTION
**Pull request by kingofkosmos**
Shows a message to the player if they're buckled to a chair/bed (with 5 second cooldown), this helps the completely new players not get stuck on the arrivals shuttle anymore. Also adds consistency since some situations in the game already show a message like this when you're restricted and try to leave the situation. Thanks to everyone who helped me out with this.

Closes #30239

### CODE REVIEW

**Comment by XDTM** [comment]
This might be very spammy if someone holds the movement key down

**Comment by kingofkosmos** [comment]
Confirmed, you can make it spammy if you want to. Does that eat up a lot of resources or something?

**Comment by PraiseRatvar** [comment]
@kingofkosmos give it a much needed cooldown

**Comment by kingofkosmos** [comment]
5 second cooldown added.

File: code/game/objects/structures/beds_chairs/bed.dm [hunk 1]
```diff
@@ -37,6 +38,11 @@
 	else
 		return ..()
 
+/obj/structure/bed/relaymove(mob/living/user)
+	if(cooldown <= world.time)
+		cooldown = world.time + 50
+		to_chat(user, "<span class='warning'>You can't move while buckled to \the [src]!</span>")
```
Comment (Cyberboss) [inline code/game/objects/structures/beds_chairs/bed.dm]: Remove `\the `

File: code/game/objects/structures/beds_chairs/chair.dm [hunk 2]
```diff
@@ -112,6 +113,11 @@
 	else
 		rotate()
 
+/obj/structure/chair/relaymove(mob/living/user)
```
Comment (Cyberboss) [inline code/game/objects/structures/beds_chairs/chair.dm]: Don't duplicate the code, either make it on obj/structure with a variable saying it should activate on chairs and beds or make it a component (better)

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/game/objects/structures/beds_chairs/chair.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/game/objects/structures/beds_chairs/chair.dm]: COMPONENTTTTTt

**Comment by PopNotes** [comment]
Okay, THIS one I support. This is how you make the game less obtuse without holding hands.

**Comment by kingofkosmos** [comment]
Changed the thing a lot, message works even better now, with tesla coil buckling, and I tested it doesn't cause any problems with the secway and janicart buckling.

**Comment by Anonmare** [comment]
@optimumtact  This is the better PR tbqh

**Comment by Cyberboss** [comment]
I want to see this implement stuff from MSO's comment here: https://github.com/tgstation/tgstation/pull/30281#issuecomment-325866339

**Comment by kingofkosmos** [comment]
image
Buckled now has a custom alert icon.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**